packages/atlas/src/views/studio/YppDashboard/YppDashboard.tsx

```diff
@@ -46,6 +74,213 @@ export const YppDashboard: FC = () => {
 
   const tiersTooltip = atlasConfig.features.ypp.tiersDefinition?.tiersTooltip
 
+  const mappedTabs = TABS.map((tab) => ({ name: tab }))
+
+  const [openModal, closeModal] = useConfirmationModal()
+
+  // todo
+  const handleLeaveTx = useCallback(() => {
+    displaySnackbar({
+      title: 'You left the progam',
+      description:
+        'You are no longer member of the YouTube Partner Program. You can now connect your YouTube channel with another Joystream channel.',
+      iconType: 'success',
+    })
+    closeModal()
+  }, [closeModal, displaySnackbar])
+
+  const handleChangeSettings = useCallback(async () => {
+    if (!currentChannel) return
+    try {
+      const data = await axios.put(
+        `${atlasConfig.features.ypp.youtubeSyncApiUrl}/channels/${currentChannel.joystreamChannelId}`,
+        {
+          category,
+          isSync,
+        }
+      )
+
+      if (data.status === 200) {
+        displaySnackbar({
+          title: 'Settings updated successfully',
+          description: 'Your videos will no longer be synced with your YouTube channel.',
+          iconType: 'success',
+        })
+        refetchSyncedChannels()
+      }
+    } catch (e) {
+      SentryLogger.error('Error while updating YPP setting: ', e)
+    }
+  }, [currentChannel, category, isSync, displaySnackbar, refetchSyncedChannels])
+
+  const content = useMemo(() => {
+    const openModalOptions = {
+      title: 'Leave the program?',
+      headerIcon: <StyledSvgAlertsWarning32 />,
+      primaryButton: {
+        text: 'Leave the program',
+        variant: 'destructive' as const,
+        onClick: () => handleLeaveTx(),
+      },
+      secondaryButton: {
+        text: 'Cancel',
+        variant: 'secondary' as const,
+        onClick: () => closeModal(),
+      },
+      description:
+        'Are you sure you want to leave the program? You will no longer receive rewards for performing the tasks, and your future YouTube videos will not be imported automatically to Joystream. ',
+    }
+
+    switch (TABS[currentVideosTab]) {
+      case 'Dashboard':
```

Comment:
I would create a separate component for the Dashboard and for Settings and keep them in separate files. wdyt?